### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by harshi0102**
Blog App - processing data in models project done according to microverse's requirements which include:

Create models
Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png).
Set up associations between models.
Remember that author_id column in posts table should be the foreign_key for the users table.
Use models to insert data
Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
Open rails console for your project.
Create at least one user by running the following code:
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
Create at least 4 posts written by one of the users you created by running the following code:
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
Create at least 6 posts comments for one of the posts you created by running the following code:
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.
Create custom methods
Your models should include some additional methods.

Users
A method that returns the 3 most recent posts for a given user.
Posts
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
Comments
A method that updates the comments counter for a post.
Likes
A method that updates the likes counter for a post.
Go to rails c and check if your methods are working.
Need a big picture?
Remind me about [the big picture of this project](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md).

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id', dependent: :destroy
-  has_many :comments, dependent: :destroy
-  has_many :likes, dependent: :destroy
```
Comment (zainsadaqat) [inline app/models/user.rb]: Dear @harshi0102 
Good job for adding the user's relation with posts However, there're two relations that are missing. As you can see in the ERD diagram. The user also has many Likes and User has many Comments. Kindly implement that as well 👍 

Screenshot 2023-07-19 at 10 21 46 PM

File: app/models/post.rb [hunk 2]
```diff
@@ -1,5 +1,17 @@
 class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User'
-  has_many :comments, dependent: :destroy
-  has_many :likes, as: :likeable, dependent: :destroy
+  belongs_to :author, class_name: 'User', foreign_key: 'user_id'
+  has_many :comments
+  has_many :likes
+
+  after_create :update_user_posts_counter
```
Comment (zainsadaqat) [inline app/models/post.rb]: ### [OPTIONAL]
Dear @harshi0102 
You can also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save

File: app/models/like.rb [hunk 3]
```diff
@@ -1,5 +1,11 @@
-# app/models/like.rb
 class Like < ApplicationRecord
-  belongs_to :user
   belongs_to :post
+  after_create :update_post_likes_counter
+  after_destroy :update_post_likes_counter
```
Comment (zainsadaqat) [inline app/models/like.rb]: [OPTIONAL]
Dear @harshi0102
You can also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
## Hello @harshi0102  👋 🙂

You're doing a great job on your project. We're proud of you. Good job so far! 

It's OK to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job. 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Flow
  ✅  Descriptive Pull Request
  ✅  No linter errors

## Required Changes ♻️

- [ ] 



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @harshi0102, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well implementing the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**